python/cudf/dataframe/string.py

```diff
@@ -521,3 +521,29 @@ def _replace_defaults(self):
     def copy(self, deep=True):
         params = self._replace_defaults()
         return type(self)(**params)
+
+    def fillna(self, fill_value, inplace=False):
+        """
+        Fill null values with * fill_value *
+        """
+        result = self.copy()
+        if not isinstance(fill_value, str) and \
+            not(
+                isinstance(fill_value, series.Series) and
+                isinstance(fill_value._column, StringColumn)
+                ):
+            raise TypeError("fill_value must be a string or a string series")
+
+        # replace fill_value with nvstrings
+        # if it is a column
+
+        if isinstance(fill_value, series.Series):
+            if len(fill_value) < len(result):
+                raise ValueError("fill value series must be of same or "
+                                 "greater length than the series to be filled")
+
+            fill_value = fill_value[: len(self)]._column._data
+
+        result._data = result._data.fillna(fill_value)
```

Comment:
Does this operation happen inplace in nvstrings? Otherwise we shouldn't need to make a copy of the data as this will return by copy anyway.

Reply:
Fixed in commit . https://github.com/rapidsai/cudf/pull/1510/commits/f20d5c5325f2fa8197bc82057c90154b3d36c156. 

We no longer create a copy of the data. I hope this resolves it.